### DISCUSSION TITLE
Search performance

### DISCUSSION THREAD

**Discussion by rjjan** [fuma-nama/fumadocs#1208] [Q&A]
I use build-in search as the tutorial example.
when the first time to use search box, the UI is spining
it seems building index ? (i don't know, there is no message in yarn dev console)
and it took long time before i can navigate to other pages.
Is it normal ?
<img width="124" alt="Screen Shot 2024-12-31 at 4 40 26 PM" src="https://github.com/user-attachments/assets/1ff123c3-2052-448d-a6af-951fbf11e893" />

how can i improve the first time search user experience ?  Thanks!!

**Comment by rjjan** [comment]
Another issue I am facing is when yarn build, I notice it iterate all the .mdx files 

I don't have many .mdx files (~ 12 files) size as attached image
<img width="662" alt="Screen Shot 2024-12-31 at 5 26 30 PM" src="https://github.com/user-attachments/assets/d8978ff7-3aec-43f2-81e6-9e44c989bb5f" />

Each time I yarn build I always wait and then get the following messages

Failed to build /[lang]/docs/[...slug]/page: /zh-TW/docs/test/by-date/2024-12-26 (attempt 1 of 3) because it took more than 60 seconds. Retrying again shortly.

By the way, I am using mac M1.

**Comment by fuma-nama** [comment]
It is necessary for server to index them (for Vercel, serverless environment means they are re-indexed when the server remounted), otherwise you will need 3rd party solutions so that the cloud service provider can index them in their own database

**Comment by fuma-nama** [comment]
The build performance issue requires a reproduction, it is working fast on our site and some large sites like Arktype docs. I think it is probably caused by some custom code in `page.tsx` or MDX content.
